pkg/operation/botanist/component/metricsserver/metrics_server.go

```diff
@@ -352,12 +352,12 @@ func (m *metricsServer) computeResourcesData() (map[string][]byte, error) {
 							},
 							Resources: corev1.ResourceRequirements{
 								Requests: corev1.ResourceList{
-									corev1.ResourceCPU:    resource.MustParse("50m"),
-									corev1.ResourceMemory: resource.MustParse("150Mi"),
+									corev1.ResourceCPU:    resource.MustParse("20m"),
+									corev1.ResourceMemory: resource.MustParse("40Mi"),
 								},
 								Limits: corev1.ResourceList{
-									corev1.ResourceCPU:    resource.MustParse("500m"),
-									corev1.ResourceMemory: resource.MustParse("1Gi"),
+									corev1.ResourceCPU:    resource.MustParse("20m"),
+									corev1.ResourceMemory: resource.MustParse("40Mi"),
```

Comment:
is it safe to decrease metrics-server container's limits from `500m` to `20m` and from `1Gi` to `40Mi`? If yes, can you explain why and what kind of testing was performed?

Reply:
With the use of `addon-resizer`, the default values for `resources` (`requests` and `limits`) are irrelevant because the addon-resizer computes and overrides (both `requests` and `limits`) based on cluster size (number of nodes if `use-metrics` is set to `false`). Also `addon-resizer` doesn't support different values for `requests` and `limits`. So, with #4216, the actual applicable base `requests` and `limits` were `cpu: 20m` and `memory: 15Mi`.

The `resources` `requests`, `limits` and `addon-resizer` configuration was based on `metrics-server` [documentation](https://github.com/kubernetes-sigs/metrics-server#scaling) and [manifests](https://github.com/kubernetes-sigs/metrics-server/blob/master/manifests/autoscale/patch.yaml).

Regarding testing, I created a fresh shoot with one node and then scaled the nodes to `5` and `10` to see that the `requests` and `limits` were scaled to `cpu: 30m` (`20m + 10 * 1m`) and `memory: 60Mi` (`40Mi + 10 * 2Mi`). The `metrics-server` was stable throughout with the resource usage around `cpu: 6m` and `memory: 30Mi`. With the base memory of `15Mi`, I could see that the `metrics-server` was getting `OOMKilled` sometimes.

Reply:
Ok, thanks for clarifying.